tests/framework/filters/AccessControlTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace yii\filters;
+
+use Yii;
+use yiiunit\TestCase;
+use Prophecy\Argument;
+use yii\filters\AccessControl;
+use yii\web\User;
+use yii\filters\AccessRule;
+
+/**
+ * @group filters
+ */
+class AccessControlTest extends TestCase
+{
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $config = [
+            'components' => [
+                'user' => [
+                    'identityClass' => 'identityClass'
+                ]
+            ]
+        ];
+        
+        $this->mockWebApplication($config);
+
+    }
+
+    public function testInitRuleIsArrayLoadedRule()
+    {
+        $config = [
+            'rules' => [
+                ['allow' => false]
+            ]
+        ];
+
+        $filter = new AccessControl($config);
+
+        $expectedFilters = [Yii::createObject(array_merge($filter->ruleConfig, ['allow' => false]))];
+
+        $this->assertEquals($expectedFilters, $filter->rules);
+    }
+
+    public function testInitRuleNotArrayEmptyRules()
+    {
+        $filter = new AccessControl();
+
+        $this->assertCount(0, $filter->rules);
+    }
+
+    public function testInitDenyCallbackEqualsNotFunctionGenerateException()
+    {
+        $this->setExpectedException('yii\base\InvalidConfigException');
+
+        new AccessControl(['denyCallback' => 'Invalid function']);
+    }
+
+    public function testInitDenyCallbackIsClosure()
+    {
+        $filter = new AccessControl(['denyCallback' => function () {}]);
+
+        $this->assertInstanceOf('\Closure', $filter->denyCallback);
```

Comment:
this test is error prone. Without the specified config, the assertion still applies.